Dynatrace/dynatrace-operator
# Create new v1beta3 dynakube CRD version

## Description

Issue [K8S-10079](https://dt-rnd.atlassian.net/browse/K8S-10079)

This PR introduce new dynakube version `v1beta3` but we still gonna store everything as `v1beta2` (as hub):

```mermaid
flowchart LR

A[v1beta1] -->B(v1beta2 HUB)
C[v1beta3]-->B(v1beta2 HUB)
```

During converation between (hub) version `v1beta2` and new version `v1beta3` i'm using annotations to store new attributes from `.spec`.

## How can this be tested?

- Deploy v1beta1 make sure all working:
```

```
- Deploy v1beta2 make sure all working

- Deploy v1beta3 make sure all working (check that you have annotations; and inside logs you see that extensions reconciled)
